gradle/validate-annotations.gradle

```diff
@@ -0,0 +1,56 @@
+import java.nio.file.Files
+
+for (def sourceSet in [sourceSets.main, sourceSets.client]) {
+	// We have to capture the source set name for the lazy string literals,
+	// otherwise it'll just be whatever the last source set is in the list.
+	def sourceSetName = sourceSet.name
+	def taskName = sourceSet.getTaskName('validate', 'Annotations')
+	def task = tasks.register(taskName, ValidateAnnotations) {
+		group = 'fabric'
+		description = "Validate annotations for $sourceSetName packages."
+
+		// Only apply to default source directory since there's also generated package-info files.
+		sourceRoot = file("src/$sourceSetName/java")
+	}
+	tasks.check.dependsOn task
+}
+
+import static groovy.io.FileType.FILES
+
+class ValidateAnnotations extends SourceTask {
+	private static final def API_STATUS_INTERNAL = ~/@ApiStatus\.Internal/
+	private static final def ENVIRONMENT = ~/@Environment/
+
+	@SkipWhenEmpty
+	@InputDirectory
+	final DirectoryProperty sourceRoot = project.objects.directoryProperty()
```

Comment:
This isnt needed when using `SourceTask`. 

You can use getSource https://github.com/gradle/gradle/blob/master/subprojects/core/src/main/java/org/gradle/api/tasks/SourceTask.java#L58

Reply:
Thanks. How about now?